src/jobs/job.rs

```diff
@@ -129,23 +124,104 @@ impl Job {
-            analyzer: Some(AnalyzerRef::Standard),
-            ignored_lines: None,
-            ignore: Default::default(),
+            ..Default::default()
+        }
+    }
+    pub fn allow_failures(&self) -> bool {
+        self.allow_failures.unwrap_or(false)
+    }
+    pub fn allow_warnings(&self) -> bool {
+        self.allow_warnings.unwrap_or(false)
+    }
+    pub fn background(&self) -> bool {
+        self.background.unwrap_or(true)
+    }
+    pub fn default_watch(&self) -> bool {
+        self.default_watch.unwrap_or(true)
+    }
+    pub fn expand_env_vars(&self) -> bool {
+        self.expand_env_vars.unwrap_or(true)
+    }
+    pub fn need_stdout(&self) -> bool {
+        self.need_stdout.unwrap_or(false)
+    }
+    pub fn extraneous_args(&self) -> bool {
+        self.extraneous_args.unwrap_or(true)
+    }
+    pub fn show_changes_count(&self) -> bool {
+        self.show_changes_count.unwrap_or(false)
+    }
+    pub fn grace_period(&self) -> Period {
+        self.grace_period
+            .unwrap_or(std::time::Duration::from_millis(15).into())
+    }
+    pub fn on_change_strategy(&self) -> OnChangeStrategy {
+        self.on_change_strategy
+            .unwrap_or(OnChangeStrategy::WaitThenRestart)
+    }
+    pub fn apply(
+        &mut self,
+        job: &Job,
+    ) {
+        if let Some(b) = job.allow_failures {
+            self.allow_failures = Some(b);
+        }
+        if let Some(b) = job.allow_warnings {
+            self.allow_warnings = Some(b);
+        }
+        if let Some(v) = job.analyzer {
+            self.analyzer = Some(v);
+        }
+        if let Some(b) = job.apply_gitignore {
+            self.apply_gitignore = Some(b);
+        }
+        if let Some(b) = job.background {
+            self.background = Some(b);
+        }
+        if !job.command.is_empty() {
+            self.command.clone_from(&job.command);
+        }
+        if let Some(b) = job.default_watch {
+            self.default_watch = Some(b);
+        }
+        for (k, v) in &job.env {
+            self.env.insert(k.clone(), v.clone());
+        }
+        if let Some(b) = job.expand_env_vars {
+            self.expand_env_vars = Some(b);
+        }
+        if let Some(b) = job.extraneous_args {
+            self.extraneous_args = Some(b);
+        }
+        for v in &job.ignore {
+            if !self.ignore.contains(v) {
+                self.ignore.push(v.clone());
+            }
+        }
+        if let Some(v) = job.ignored_lines.as_ref() {
+            self.ignored_lines = Some(v.clone());
+        }
+        if let Some(v) = job.kill.as_ref() {
+            self.kill = Some(v.clone());
+        }
+        if let Some(b) = job.need_stdout {
+            self.need_stdout = Some(b);
+        }
+        if let Some(v) = job.on_change_strategy {
+            self.on_change_strategy = Some(v);
+        }
+        if let Some(v) = job.on_success.as_ref() {
+            self.on_success = Some(v.clone());
+        }
```

Comment:
There doesn't seem to be code to merge grace_period

Reply:
Good find!

I'll add a unit test creating 2 instances of Job, without using the Default implementation, to check there's nothing missing. Or I'm sure I'll add regressions later on...

Reply:
In another project I use macros for the same exact cases, but that other project takes minutes to compile ^^

Reply:
I fixed it, with the unit test. There might still be traps, eg on not being able to set some default-looking values.